### DISCUSSION TITLE
Action in Edit

### DISCUSSION THREAD

**Discussion by weweb-hu** [filamentphp/filament#7788] [Help] [answered]
Hey,

I am quite a newbie in filament, so please accept that I cant sort this one out. I might be tired too, as this issue has been ruining my day since 8 am...

So, I am trying to make an action, which does nothing, but sends an email to the actual Order, that I have been editing.

The code looks like this:
```
namespace App\Filament\Resources\OrderResource\Pages;

use App\Models\Order;
use App\Mail\VoucherMail;
use Filament\Pages\Actions;
use Illuminate\Support\Facades\Mail;
use Filament\Resources\Pages\EditRecord;
use App\Filament\Resources\OrderResource;

class EditOrder extends EditRecord
{
    protected static string $resource = OrderResource::class;

    protected function getActions(): array
    {
        return [
            Actions\Action::make('sendEmail')
                ->label('Send e-mail voucher')
                ->color('danger')
                ->action(function (array $arguments) {
                    $recipient = Order::find($arguments['email']);
                    Mail::to($recipient)
                        ->send(new VoucherMail());
                }),
            Actions\DeleteAction::make(),
        ];
    }
}

```

I simply can not sort out, what's wrong as it doesnt reach the recipient email address. The next issue would be to pass all the record data to the email, as it would need to contain.

I hope I was understandable.

**Comment by leandrocfe** [comment]
```php
Actions\Action::make('sendEmail')
    ->label('Send e-mail voucher')
    ->color('danger')
    ->action(function ($record) {
        $recipient = Order::find($record->email);
        Mail::to($recipient)->send(new VoucherMail());
}),
```

**Reply by weweb-hu** [reply]
I tested, unfortunately it came back with an error:
ErrorException - Attempt to read property "email" on null

**Reply by leandrocfe** [reply, marked as answer]
what about `$this->record->email` ?

**Reply by weweb-hu** [reply]
It seems to be working as it get back with the error which should contain the passed data (from the mail). What should I pass to the view if I would like to paste data there? Btw I really appreciate your help. I have been checking filament documentation half day and I couldn't sort out the solution for this... do you think if there is a part about this?

**Reply by danharrin** [reply]
I assume you're using v2? In v3, `$record` should be available as per Leandro's original message
